syntax = "proto3";

// fix this with compiler flag -I
import "common.proto";

option java_generic_services = true;
option java_package = "com.example.beamyconfiguration";
option java_outer_classname = "Functional";

package base;



service FunctionalService {
  rpc OpenPassWindow (ClientId) returns (Empty) {}
  rpc ClosePassWindow (ClientId) returns (Empty) {}
  rpc SetFanSpeed (SenderInfo) returns (Empty) {}
  rpc SubscribeToFanSpeed (SubscriberRequest) returns (stream Value) {}
}

// to stop hammering make same call with frequency 0
message SenderInfo {
  ClientId clientId = 1;
  Value value = 2;
  int32 frequency = 3;
}

message SubscriberRequest {
  ClientId clientId = 1;
  bool onChange = 2;
}

message Value {
  int32 payload = 1;
}
